doc/source/data/working-with-pytorch.rst

```diff
@@ -0,0 +1,505 @@
+.. _working_with_pytorch:
+
+Working with PyTorch
+====================
+
+Ray Data integrates with the PyTorch ecosystem.
+
+This guide describes how to:
+
+* :ref:`Iterate over your dataset as torch tensors for model training <iterating_pytorch>`
+* :ref:`Write transformations that deal with torch tensors <transform_pytorch>`
+* :ref:`Perform batch inference with torch models <batch_inference_pytorch>`
+* :ref:`Save Datasets containing torch tensors <saving_pytorch>`
+* :ref:`Migrate from PyTorch Datasets to Ray Data <migrate_pytorch>`
+
+.. _iterating_pytorch:
+
+Iterating over torch tensors for training
+-----------------------------------------
+To iterate over batches of data in torch format, call :meth:`Dataset.iter_torch_batches() <ray.data.Dataset.iter_torch_batches>`. Each batch is represented as `Dict[str, torch.Tensor]`, with one tensor per column in the dataset. 
+
+This is useful for training torch models with batches from your dataset. See `the API reference <ray.data.Dataset.iter_torch_batches>` for more configuration details such as providing a `collate_fn` for customizing the conversion.
+
+.. testcode::
+
+    import ray
+    import torch
+
+    ds = ray.data.read_images("example://image-datasets/simple")
+
+    for batch in ds.iter_torch_batches(batch_size=2):
+        print(batch)
+
+.. testoutput::
+    :options: +MOCK
+
+    {'image': tensor([[[[...]]]], dtype=torch.uint8)}
+    ...
+    {'image': tensor([[[[...]]]], dtype=torch.uint8)}
+
+Integration with Ray Train
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Ray Data also integrates with :ref:`Ray Train <train-docs>` for easy data ingest for data parallel training, with support for PyTorch, PyTorch Lightning, or Huggingface training.
+
+See the :ref:`Ray Train user guide <train-datasets>` for more details.
```

Comment:
In the other guides and "Getting started", we put "For more details..." after the code snippet. Should we do the same here? Tradeoff I see is: 

- Advantage: Looks cleaner
- Disadvantage: Cross reference less visible

Don't have a strong opinion but consistency is nice

Reply:
Updated to adding references below the code snippets.